### ISSUE TITLE
Shall we set tenant only for new records

### ISSUE THREAD

**Issue by crccw** [PerfectMemory/mongoid-multitenancy#18]
Currently, if current_tenant is set, all records, whether newly created ones or already persisted ones, will have tenant set to current_tenant. Is this the indented behavior? This is especially strange if optional is set to true..

**Comment by crccw** [comment]
For myself, only set tenant to current_tenant for new record is wanted. I've created a pull request for it. #19

**Comment by abrisse** [comment]
@crccw  No you're right. I already fixed it locally yesterday with the same changes that you did. I will push a new version today with new specs. Thanks for your pull request anyway. :)

The only odd behavior that stays is the following one:

```ruby
article = Article.new(:title => 'New article', :client => nil)
article.save # => <#Article _id: 50ca04b86c82bfc125000044, title: 'New blog', client_id: 50ca04b86c82bfc125000025>
article.tenant = nil
article.save => <#Article _id: 50ca04b86c82bfc125000044, title: 'New blog', client_id: nil>
end
```

It means that `Article.new(title: 'New article', client: nil)` <=> `Article.new(:title => 'New article')`

I cannot change this behaviour without hacking mongoid. Are you okay with it?

**Comment by abrisse** [comment]
Fixed in 81a9b45a34a49f26d17cc1cdf44fd7699bad20f9

**Comment by crccw** [comment]
It's ok for me. Seems no way to fix it unless we have a before_initialize callback..
